### ISSUE TITLE
IDE0055 is not reported for tabs instead of spaces after a single line comment in collection initializers

### ISSUE THREAD

**Issue by arphox** [dotnet/roslyn#71143]
**Version Used**: I'm not sure how to check it, `dotnet --version` yields `8.0.100`.

**Issue**:
![kép](https://github.com/dotnet/roslyn/assets/17320094/80d4755f-fe71-4407-9678-380996f159d7)

**Steps to Reproduce**:
Source code:
```cs
namespace Demo;

public static class Program
{
    public static void Main()
    {
        List<MyClass> objs = new()
        {
            new MyClass
            {
                MyProp = "obj1", // this is a comment
			},
        };

        var obj2 = new MyClass
        {
            MyProp = "obj2", // this is a comment
		};

        Console.WriteLine(objs.Count);
    }
}

class MyClass
{
    public string? MyProp { get; set; }
}
```

csproj:
```cs
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>net8.0</TargetFramework>
        <ImplicitUsings>enable</ImplicitUsings>
        <Nullable>enable</Nullable>
        <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
        <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
    </PropertyGroup>
</Project>
```

.editorconfig:
```
[*]
charset = utf-8
indent_style = space
indent_size = 4
trim_trailing_whitespace = true
insert_final_newline = true

dotnet_diagnostic.IDE0055.severity = warning
```

**Diagnostic Id**: IDE0055

**Expected Behavior**: IDE0055 issue reported for Line 12, because spaces must be used.

**Actual Behavior**: **no** IDE0055 issue reported for Line 12
